appcues/src/main/java/com/appcues/ui/presentation/ViewPresenter.kt

```diff
@@ -20,7 +25,14 @@ internal abstract class ViewPresenter(
 ) : DefaultLifecycleObserver {
 
     private lateinit var viewModel: AppcuesViewModel
-    private lateinit var gestureListener: ShakeGestureListener
+
+    private val experienceRenderer: ExperienceRenderer = scope.get()
+    private val appcuesCoroutineScope: AppcuesCoroutineScope = scope.get()
+
+    private var gestureListener: ShakeGestureListener? = null
+
+    private val currentExperience: Experience?
+        get() = experienceRenderer.getState(renderContext)?.currentExperience
```

Comment:
we're able to lookup the current experience by `RenderContext` now, and this enables us to tell if the trigger was a `Preview`, as well as get the `id` for refresh, on shake.